### PR TITLE
Add initial check-cla action.

### PR DESCRIPTION
**Pull request by jezdez**
This is supposed to implement the initial features currently in use by with anaconda-issue-bot as documented in https://github.com/conda/infra/issues/538.

Fixes https://github.com/conda/infra/issues/548.

### CODE REVIEW

File: check-cla/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    if: >-
+      !github.event.repository.fork
+      && (
+        github.event.comment.body == '@conda-bot check'
+        || github.event_name == 'pull_request'
+      )
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check CLA
+        uses: conda/actions/check-cla@check-cla
```
Comment (jezdez) [inline check-cla/README.md:35]:
```suggestion
        uses: conda/actions/check-cla
```

Before this is merged we need to make this change.

**Comment by jezdez** [comment]
Hmm, this doesn't correctly handle when leaving the "@conda-bot check" comment after the initial run since that runs in a workflow run that isn't tied to the pull request anymore and won't correctly update the status of the PR on success. I'm thinking closing and reopening the PR may suffice to trigger the actual recheck instead of listening to comments.

Here's the test repo: https://github.com/conda/test-cla

@kenodegard @travishathaway Any opinions on this?

File: check-cla/action.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          const creator = context.payload.pull_request.user.login;
+          const hasSigned = contributors.includes(creator);
+          console.log(context);
+          console.log(contributors);
+          console.log(hasSigned);
+          core.setOutput('contributors', contributors);
+          core.setOutput('hasSigned', hasSigned);
+
+    # add cla-signed label if actor has already signed
+    - name: Add label
```
Comment (kenodegard) [inline check-cla/action.yml:45]: Remove label if we find that CLA is unsigned?

File: canary-release/README.md [hunk 3]
```diff
@@ -46,10 +46,6 @@ jobs:
       - name: Create and upload canary build
         uses: conda/actions/canary-release
         with:
-          # [required]
-          # the pull request ID
-          pr: ${{ github.event.number }}
-
```
Comment (kenodegard) [inline canary-release/README.md:52]: Separate PR?

File: check-cla/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+on:
+  issue_comment:
+    types:
+      - created
+  pull_request_target:
+    types:
+      - reopened
+      - opened
+      - closed
```
Comment (kenodegard) [inline check-cla/README.md:21]: I don't think we care about the `closed` event?

File: check-cla/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      - synchronize
+
+jobs:
+  check:
+    if: >-
+      !github.event.repository.fork
+      && (
+        github.event.comment.body == '@conda-bot check'
+        || github.event_name == 'pull_request_target'
```
Comment (kenodegard) [inline check-cla/README.md:30]: We don't care about labeling issues as `cla-signed` so we should also check that the `issue_comment` event is occurring on PRs

```suggestion
        github.event.comment.body == '@conda-bot check'
        && github.event.issue.pull_request
        || github.event_name == 'pull_request_target'
```

**Comment by travishathaway** [comment]
@jezdez , I tested this out again by submitting a new PR https://github.com/conda/test-cla/pull/7  It looks like everything is working.

**Comment by jezdez** [comment]
I think this should be good to go, finally!

**Pull request merged**